src/Chapter2.hs

```diff
@@ -857,8 +879,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
-
+rewind :: [a] -> [a]
+rewind [] = []
+rewind (x:xs) = rewind xs ++ [x]
```

Comment:
Usually, adding elements to the end of the list is costly. Your implementation is correct :+1: But you can try to implement this function more efficiently by using `:` instead `++`. If you can think, how to avoid adding elements to the end of the list, and instead add to the beginning, but get the same result, that would be amazing!

In general, it's always a good idea to try to replace `l ++ [x]` with `x : l` whenever possible, though, in some cases it might not be that trivial to achieve the same result. Remember the analogy with trains 🚂 : :train: :train: :train: